corn-maze.lic

```diff
@@ -0,0 +1,632 @@
+      drop_item(item)
+    end
+  end
+
+  def wait_on_done_move
+    last_room = XMLData.room_count
+    pause 0.5
+    while last_room != XMLData.room_count
+      last_room = XMLData.room_count
+      pause 0.5
+    end
+  end
+
+  def get_task
+    wait_on_done_move
+    @halfling_path.each do |x|
+      exit if Flags['maze_done']
+      if DRRoom.npcs.include? 'Halfling'
+        break
+      else
+        custom_move "#{x}"
+      end
+    end
+    fput 'ask halfling about task'
+    echo "Task count = #{@task_count}"
+    if @searching == false
+      case bput('ask halfling for task', 'A cheerful Halfling smiles happily', 'To whom are you', 'A cheerful Halfling looks at you and says')
+      when 'A cheerful Halfling looks at you and says'
+        fput('ask halfling for task cancel')
+        fput('ask halfling for task cancel')
+        return get_task
+      end
+    end
+    @done = false
+    find_start
+    @done = false
+    wait_for_command
+  end
+
+  def find_start
+    wait_on_done_move
+    unless XMLData.room_title.include?('Labyrinth') || XMLData.room_title.include?('Corn Maze, Entrance')
+      DRC.message("The script can't find its way back from here!  Make your way back towards the northwest Labyrinth using elanthipedia's Corn Maze entry, such as https://elanthipedia.play.net/Sleeping_Dragon_Corn_Maze_432 and then unpause the script!")
+      pause_script('cornmaze')
+    end
+    until @done
+      exit if Flags['maze_done']
+      find_exits = DRC.bput("look",/Obvious paths: .*\./,)
+      exits = find_exits.match(/^Obvious paths: (.*)\./).to_a[1].split(', ')
+      if (exits.include? 'north')
+        custom_move 'north'
+      elsif (exits.include? 'east')
+        custom_move 'east'
+      elsif (exits.include? 'south')
+        custom_move 'south'
+        custom_move 'south'
+        custom_move 'east'
+      elsif exits.include?('west') && !exits.include?('southwest') && !exits.include?('east') && !exits.include?('south') && !exits.include?('north') && !exits.include?('southeast') && !exits.include?('northwest') && !exits.include?('northeast')
+        custom_move 'west'
+        custom_move 'west'
+      elsif ((exits.include? 'west') && (exits.include? 'southwest'))
+        custom_move 'west'
+        custom_move 'west'
+        custom_move 'west'
+        custom_move 'west'
+        @done = true
+      end
+    end
+  end
+
+  def finish_task
+    reset_flags
+    @task_count = @task_count + 1
+    get_task
+  end
+
+  def reset_flags
+    Flags.reset('scarecrow')
+    Flags.reset('mice')
+    Flags.reset('weeds')
+    Flags.reset('traps')
+    Flags.reset('grasshoppers')
+    Flags.reset('scream')
+    Flags.reset('landmarks')
+    Flags.reset('tokens')
+    Flags.reset('corn')
+    Flags.reset('halflings')
+    Flags.reset('loot')
+  end
+
+  def custom_move(dir, fast=false)
+    if fast && $frontend != 'genie'
+      put dir
+      @current_type_ahead += 1
+      if @type_ahead_size < @current_type_ahead
```

Comment:
💡 Why is this conditional in here if `@current_type_ahead` always starts at 0 and `@type_ahead_size` is always 1?

Reply:
Looks like once upon a time they intended to make that adjust itself.  I don't think we'd ever need it to be more than 1 extra though, so I suppose the extra steps could be eliminated